pkg/fleets/controller.go

```diff
@@ -461,18 +477,53 @@ func (c *Controller) rollingUpdateActive(fleet *agonesv1.Fleet, active *agonesv1
 }
 
 // rollingUpdateRest applies the rolling update to the inactive GameServerSets
-func (c *Controller) rollingUpdateRest(fleet *agonesv1.Fleet, rest []*agonesv1.GameServerSet) error {
+func (c *Controller) rollingUpdateRest(fleet *agonesv1.Fleet, active *agonesv1.GameServerSet, rest []*agonesv1.GameServerSet) error {
```

Comment:
For rollingUpdateActive and rollingUpdateRest - I'm wondering if we create duplicate functions with the new functionality for the new feature gated strategies, and have a check at the top in rollingUpdateDeployment (maybe rollingUpdateRestOnRead & rollingUpdteActiveOnReady).

I say this because this code was quite complicated to start, and I'm finding it hard to wrap my head around each of the implications of turning on or off the feature branch while it's interwoven in these functions. 

Also this way, there is far less chance of accidentally including a bug, as they are totally seperate code blocks. WDYT?

Reply:
Thanks for the review, let me check this tomorrow. Miss this notification.

Reply:
Coming back to this - out of all the comments, this is the big blocker for me to get this through.

Reply:
I have resolved your comment. Now old and a new one are not interwoven. Thanks for the hint. This actually helps and with debugging also.